lib/hooks/use-products-list.ts

```diff
@@ -0,0 +1,54 @@
+import { SerializeFrom } from '@remix-run/node';
+import { useEffect, useState } from 'react';
+import { getEcomApi, Product } from '../ecom';
+import { useAppliedProductFilters } from './use-applied-product-filters';
+import { useProductSorting } from './use-product-sorting';
+
+export function useProductsList(
+    categorySlug: string,
+    initialProducts: (Product | SerializeFrom<Product>)[],
+    initialTotalCount: number,
+) {
+    const [products, setProducts] = useState(initialProducts);
+    useEffect(() => {
+        setProducts(initialProducts);
+    }, [initialProducts]);
+
+    const [totalProductsCount, setTotalProductsCount] = useState(initialTotalCount);
+    useEffect(() => {
+        setTotalProductsCount(initialTotalCount);
+    }, [initialTotalCount]);
+
+    const [isLoadingProducts, setIsLoadingProducts] = useState(false);
+
+    const api = getEcomApi();
+    const { appliedFilters } = useAppliedProductFilters();
+    const { productsSortBy } = useProductSorting();
+
+    const loadMore = async () => {
+        setIsLoadingProducts(true);
+
+        try {
+            const nextProductsResponse = await api.getProductsByCategory(categorySlug, {
+                filters: appliedFilters,
+                sortBy: productsSortBy,
+                skip: products.length,
+            });
+
+            if (nextProductsResponse.status === 'success') {
+                setProducts((prev) => [...prev, ...nextProductsResponse.body.items]);
+                setTotalProductsCount(nextProductsResponse.body.totalCount);
+            }
+        } finally {
```

Comment:
maybe add `catch` with just `console.log` or `alert` for now.. and once toasts functionality is merged we'll show the toast.

Reply:
done

Reply:
of course it's temporary, but maybe alert is better.. at least user will see something